utils/zerovec/src/map/hashmap.rs

```diff
@@ -0,0 +1,270 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::*;
+use crate::ule::AsULE;
+use crate::ZeroVec;
+use ahash::AHasher;
+use alloc::borrow::Borrow;
+use alloc::vec::{from_elem, Vec};
+use core::hash::{Hash, Hasher};
+
+#[inline]
+fn create_hasher_with_seed(seed: u128) -> AHasher {
+    AHasher::new_with_keys(seed, 0xaabbccdd)
+}
+
+#[inline]
+fn compute_hash<K: Hash>(seed: u32, k: K, m: usize) -> u32 {
+    let mut hasher = create_hasher_with_seed(seed.into());
+    k.hash(&mut hasher);
+    (hasher.finish() % m as u64) as u32
+}
+
+#[derive(Debug, Clone)]
+pub struct HashIndex<'a> {
+    displacements: ZeroVec<'a, u32>,
+    reverse_mapping: ZeroVec<'a, u32>,
+}
+
+impl<'a> HashIndex<'a> {
+    #[inline]
+    pub fn build_from_exact_iter<K, I, A>(iter: I) -> Self
+    where
+        A: Borrow<K>,
+        K: 'a + ?Sized + Hash,
+        I: ExactSizeIterator<Item = A>,
+    {
+        HashIndex::build_from_exact_iter_with_hash_fn(iter, |seed, k, len| {
+            compute_hash(seed, k, len)
+        })
+    }
+
+    #[inline]
+    pub fn build_from_exact_iter_with_hash_fn<K, I, A, H>(iter: I, h: H) -> Self
+    where
+        A: Borrow<K>,
+        K: 'a + ?Sized,
+        I: ExactSizeIterator<Item = A>,
+        H: Fn(u32, &K, usize) -> u32,
+    {
+        let iter_len = iter.len();
+
+        // A vector to track the size of buckets for sorting.
+        let mut bucket_sizes = from_elem(0, iter_len);
```

Comment:
For 'static' functions the style recommendation is to qualify them with one level of context (`vec::from_elem`).

That said, `from_elem` seems to be `#[doc(hidden)]` and I cannot figure out what the arguments are supposed to do. Is this just `vec![0; iter_len]`?

Reply:
Yes I think this is what `vec!` macro uses internally. https://doc.rust-lang.org/src/alloc/macros.rs.html#42-55

Reply:
Please use the macro then, that's what everyone knows and understands.

Reply:
I recommend importing the macro with `use alloc::vec;`. This is what we do in other places as well, and the `#[macro_use]` attribute is like a wildcard import, which is something we want to avoid.

Reply:
Done.